sys/kern/subr_vmem.c

```diff
@@ -934,6 +982,14 @@ vmem_import(vmem_t *vm, vmem_size_t size, vmem_size_t align, int flags)
 	bt_restore(vm);
 	if (error)
 		return (ENOMEM);
+#ifdef __CHERI_PURE_CAPABILITY__
+	if (vm->vm_flags & VMEM_CAPABILITY_ARENA) {
+		KASSERT(cheri_gettag(addr), ("Expected valid capability"));
+		KASSERT(cheri_getlen(addr) == size,
+		    ("Inexact bounds expected %zx found %zx",
+		    (size_t)size, (size_t)cheri_getlen(addr)));
```

Comment:
The first cast is redundant because vmem_size_t is just size_t, though maybe it should be there still to avoid leaking that implementation detail. The second shouldn't be needed because the builtin is defined as returning size_t.